cmd/client/search.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/Masterminds/squirrel"
+	pgUtils "github.com/monstercat/golib/db/postgres"
+	"github.com/monstercat/gologx/logxhost"
+	"github.com/olekukonko/tablewriter"
+)
+
+func showSearch(name string, args []string) error {
+	var postgres string
+	var service string
+	var matcher string
+	var limit int
+	var datebefore string
+	var dateafter string
+	var orderby string
+
+	limitDefault := 50
+
+	set := flag.NewFlagSet(name, flag.ExitOnError)
+	set.StringVar(&postgres, "postgres", "", "Postgres database")
+	set.StringVar(&service, "service", "", "Defines a service to be filtered")
+	set.StringVar(&matcher, "matcher", "", "Defines a matcher to be filtered. Searches machine and name")
+	set.IntVar(&limit, "limit", limitDefault, "Defines the limit of results in the list")
+	set.StringVar(&datebefore, "datebefore", "", "Filters before a specific date time")
+	set.StringVar(&dateafter, "dateafter", "", "Filters after a specific date time")
+	set.StringVar(&orderby, "orderby", "", "Orders list by a defined column")
+
+	if err := set.Parse(args); err != nil {
+		return err
+	}
+
+	if service == "" {
+		fmt.Println(styleDanger("service flag is required"))
+		os.Exit(1)
+	}
+
+	dateRegex := regexp.MustCompile(`\d{4}-\d{2}-\d{2}`)
+	if valid := dateRegex.MatchString(datebefore); datebefore != "" && !valid {
+		fmt.Println(styleDanger("yyyy/mm/dd format is required for datebefore"))
+		os.Exit(1)
+	}
+
+	if valid := dateRegex.MatchString(dateafter); dateafter != "" && !valid {
+		fmt.Println(styleDanger("yyyy/mm/dd format is required for dateafter"))
+		os.Exit(1)
+	}
+
+	db, err := logxhost.GetPostgresConnection(postgres)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+
+	search := getQuery(service, matcher, limit, datebefore, dateafter, orderby)
+
+	logs, err := logxhost.SelectLogs(db, search)
+	if err != nil {
+		return err
+	}
+
+	if len(logs) == 0 {
+		fmt.Println(styleWarning("No logs were found"))
+		os.Exit(1)
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Id", "Machine", "Service", "Log Type", "Message", "Log Time"})
+
+	for _, l := range logs {
+		row := []string{l.Id, l.Machine, l.Service, l.LogType, l.Message, l.LogTime.Format("2006-01-02 15:04:05")}
+		table.Append(row)
+	}
+
+	table.Render()
+
+	return nil
+}
+
+func getQuery(service string, matcher string, limit int, datebefore string, dateafter string, orderby string) squirrel.SelectBuilder {
+	q := logxhost.BuildSelectLogs().Where("service LIKE '" + service + "'")
+
+	if matcher != "" {
+		some := squirrel.Or{}
```

Comment:
perhaps using Ilike 
https://github.com/monstercat/golib/blob/8ec3452d9b42b3314095c899f3f97753d47d6116/db/squirrel.go

Reply:
no need append also. Can just make the variable directly.

Reply:
But we normally have all query creation code (i.e., things that have the sql) in the lib package. The client shouldn't have to deal with it. (i.e., in the logxhost package). Probably in  logxhost/log.go